src/site/_drafts/perception-toolkit/index.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Visual searching with the Perception Toolkit
+subhead: Real-world interactivity without learning artificial intelligence
+author: joemedley
+date: 2019-05-07
+hero: hero.jpg
+# You can adjust the position of your hero image with this property.
+# Values: top | bottom | center (default)
+# hero_position: bottom
+alt: A description of the hero image for screen reader users.
+description: |
+  Wouldn't it be great if users could search your site using their camera?
+tags:
+  - visual-search
+---
+Wouldn't it be great if users could search your site using their camera? Imagine
+this. Your site is Razor McShaveyface. Your customers tell you they have trouble
+finding the right cartridges for their razor when they reorder. They don't know
+the right keywords for your product search. And let's be honest, they probably
+never will.
+
+What if they never need to? What if they could point their phone's camera at a
+cartridge they already have and your site could present them with the right
+cartridge and a big red "reorder" button?
+
+Think about other scenarios. Imagine a site that supports in-store price
+checking. Imagine getting information about a museum exhibit or historical
+marker. Imagine identifying real-world landmarks in games like geocaching or
+scavenger hunts.
+
+These are the problems the Perception Toolkit solves and I'm going to introduce
+you to it.
+
+## How does it work?
+
+The open-source Web Perception Toolkit helps you add visual search to your
+website. It passes a device camera stream through a set of detectors that map
+markers or targets to structured data on your site. With this data, you can
+present the right information in a customizable UI.
+
+I'll show you enough of this to give you a taste of how it works. For a complete
+explanation, check out the Getting Started Guide, the toolkit reference, and the
+demos.
+
+## Customize the user experience
+
+Getting UI elements is amazingly simple. The toolkit gives you livecycle events,
+Card and Button objects for crafting the user experience around those events,
+and an easy way to style the cards.
+
+The most important lifecycle event is MarkerChanges, which is called every time
+a marker is found. A marker is the toolkit's term for one of the real-world
+objects you're using the toolkit to look for. A marker can be a physical object,
+an image, a barcode, or a QR code.
+
+The process for responding to the event is the same as for any other event.
+
+```js
+const container = document.querySelector('.container');
+async function onMarkerChanges(event) {
+  // Process the event.
+}
+window.addEventListener(PerceptionToolkit.Events.MarkerChanges, onMarkerChanges);
+```
+
+Let's look at the event more closely. The event itself contains arrays of
+markers it's both found and lost. This helps account for hand and marker
+movements: cameras not help steadily enough, dropped markers, that kind of
+thing. As long as the toolkit finds the marker at some point, it will be in
```

Comment:
"As long as the toolkit finds the marker at some point, it will be in
either array."

I read this sentence a few times but wasn't sure what I was meant to take away from it. It may be self evident based on the previous sentences that something will go from the found to the lost arrays.